### PR TITLE
test(create-atomic): atomic + HSP tests

### PR DESCRIPTION
**Pull request by ThibodeauJF**
Tests for this PR: https://github.com/coveo/cli/pull/728
Some tests are ignored/skipped to make the build faster, I just want to at least get the setup to work.

### When running locally:

**Console:**
Screen Shot 2022-05-25 at 1 13 10 PM

**stderr:**
```
[13:04:48]npm WARN exec The following package was not found and will be installed: @coveo/create-atomic@1.28.1
[13:04:48][13:05:16]⠋ [13:05:16]✔  -> Using the default search page template.
[13:05:16]⠋ [13:05:16]✔  ++ /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/.env
[13:05:16][13:05:16]⠋ [13:05:16]✔  ++ /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/.gitignore
[13:05:16]⠋ [13:05:16]✔  ++ /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/package.json
[13:05:16]⠋ [13:05:16]⠙ [13:05:16]⠹ [13:05:17]✔  +! 19 files added
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/src/components.d.ts
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/src/html.d.ts
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/src/index.ts
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/src/pages/index.html
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/src/style/index.css
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/src/components/results-manager/results-manager.tsx
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/src/components/sample-component/sample-component.css
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/src/components/sample-component/sample-component.tsx
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/src/components/sample-result-component/sample-result-component.css
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/src/components/sample-result-component/sample-result-component.tsx
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/scripts/clean-up.js
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/scripts/setup-lambda.js
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/scripts/utils.js
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/.env.example
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/tsconfig.json
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/stencil.config.ts
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/netlify.toml
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/README.md
[13:05:17] -> /idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project/start-netlify.mjs
[13:05:17]⠋ [13:05:17]✔  -> 1 result template(s) generated
[13:05:17]⠋ ✔  -> Installing packages...[13:09:08]✔  -> Installation complete
[13:09:08][13:09:08]⠋ ✔  -> 
[13:09:08]          To get started:
[13:09:08]          > cd idc9e3fd5c19a245bd6c6f50f88da6dccag-atomic-project
[13:09:08]          > npm run site:init (optional, sets up a new Netlify site)
[13:09:08]          > npm start
[13:09:08]  
[13:09:08]          To share your site with the world:
[13:09:08]          > npm run site:deploy
[13:09:08]      
[13:09:08]          Happy hacking!
[13:09:08]          
[13:09:08]⠋ [13:09:08]npm notice[13:09:08] 
[13:09:08]npm notice New minor version of npm available! 8.5.0 -> 8.10.0
[13:09:08]npm notice Changelog: <https://github.com/npm/cli/releases/tag/v8.10.0>
[13:09:08]npm notice Run `npm install -g npm@8.10.0` to update!
[13:09:08]npm notice 
[13:09:08]
```

**stdout:**
```
[13:05:16]? Use an existing hosted search page as a template, or start from scratch? (Use 
[13:05:16]arrow keys)
[13:05:16]❯ Start from scratch 
[13:05:16]  ──────────────
[13:05:16]  00111lvu 
[13:05:16]  QA test 2 
[13:05:16]  00000000000000testttttttttttttttttttttttttttttttttttttttttttttttttttttttttt 
[13:05:16]jfalsfj 4uerjwlsdmc,k fowrj jsdjfhkwaiwewrfajwkfjawkf 
[13:05:16]  test 1 
[13:05:16]  test 
[13:05:16]  new search page 6 
[13:05:16]  new search page 7 
[13:05:16](Move up and down to reveal more choices)[13:05:16][41D[41C[13:05:16][2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G? Use an existing hosted search page as a template, or start from scratch? Start
[13:05:16] from scratch[1A[13D[93C
[13:05:16][13:05:16][1B
```

### When running on the CI:

**Console:**
Screen Shot 2022-05-25 at 1 17 34 PM

**stdout** & **stderr** are empty!

### CODE REVIEW

File: packages/cli-e2e/__tests__/atomic.specs.ts [hunk 1]
```diff
@@ -37,6 +37,19 @@ describe('ui:create:atomic', () => {
       projectName
     );
 
+    const buildTerminalPagePromptPromise = buildTerminal
```
Comment (ThibodeauJF) [inline packages/cli-e2e/__tests__/atomic.specs.ts:40]: Probably something wrong i'm doing here... even if I can make it work locally

File: packages/cli-e2e/__tests__/atomic.specs.ts [hunk 2]
```diff
@@ -37,6 +37,12 @@ describe('ui:create:atomic', () => {
       projectName
     );
 
+    await buildTerminal
```
Comment (ThibodeauJF) [inline packages/cli-e2e/__tests__/atomic.specs.ts:40]: Probably something wrong i'm doing here... even if I can make it work locally

Comment (louis-bompart) [reply packages/cli-e2e/__tests__/atomic.specs.ts:40]: What's the problem doc?

File: packages/cli-e2e/__tests__/atomic.specs.ts [hunk 3]
```diff
@@ -111,32 +117,30 @@ describe('ui:create:atomic', () => {
...
     ];
 
-    createdFilesPaths.forEach((path) =>
-      expect(projectFileExist(path)).toBe(true)
-    );
-  });
+    createdFilesPaths.forEach((path) => {
+      it(`should create the "${path}" file or directory`, () =>
+        expect(projectFileExist(path)).toBe(true));
+    });
```
Comment (louis-bompart) [inline packages/cli-e2e/__tests__/atomic.specs.ts:132]: This no, instead https://jestjs.io/docs/api#testeachtablename-fn-timeout or https://jestjs.io/docs/api#describeeachtablename-fn-timeout

Same for all foreaches

**Review by louis-bompart — APPROVED** [review]

**Review by olamothe — APPROVED** [review]

**Review by y-lakhdar — APPROVED** [review]

**Pull request merged**